src/shims/env.rs

```diff
@@ -465,4 +465,42 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         Ok(())
     }
+
+    fn getpid(&mut self) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+        let target_os = &this.tcx.sess.target.os;
+        assert!(
+            target_os == "linux" || target_os == "macos",
+            "`getpid` is only available for the UNIX target family"
+        );
+
+        if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
+            this.reject_in_isolation("getpid", reject_with)?;
+            this.set_last_error_from_io_error(ErrorKind::PermissionDenied)?;
+
+            return Ok(-1);
+        }
```

Comment:
As [discussed on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/269128-miri/topic/getuid.20shim/near/285408386), return an error code from `getpid` violates its spec.
```suggestion
        this.check_no_isolation("`getpid`")?;
```